src/lib/task.js

```diff
@@ -589,7 +589,7 @@ class Task extends EventEmitter {
         reporter = queue.reportFailed;
         taskState = 'failed';
       }
-      let purgeStatuses = this.task.payload.onExistStatus.purgeCaches;
+      let purgeStatuses = this.task.payload.onExitStatus && this.task.payload.onExitStatus.purgeCaches;
```

Comment:
I suggest doing: `(this.task.payload.onExitStatus || {}).purgeCaches || []` this way if `purgeCaches` is null the next thing still works... 

oh, nvm, I see you have a check there already...